src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -4,26 +4,79 @@
 from pathlib import PurePath
 
 from pants.backend.python.dependency_inference.import_parser import find_python_imports
-from pants.backend.python.dependency_inference.module_mapper import PythonModule, PythonModuleOwner
+from pants.backend.python.dependency_inference.module_mapper import (
+    PythonModule,
+    PythonModuleOwner,
+    map_first_party_modules_to_addresses,
+    map_module_to_address,
+    map_third_party_modules_to_addresses,
+)
 from pants.backend.python.dependency_inference.python_stdlib.combined import combined_stdlib
-from pants.backend.python.target_types import PythonSources
+from pants.backend.python.rules import inject_ancestor_files
+from pants.backend.python.rules.inject_ancestor_files import AncestorFiles, AncestorFilesRequest
+from pants.backend.python.target_types import PythonSources, PythonTestsSources
 from pants.core.util_rules.strip_source_roots import (
     SourceRootStrippedSources,
     StripSourcesFieldRequest,
 )
 from pants.engine.fs import Digest, DigestContents
-from pants.engine.rules import rule
+from pants.engine.internals.graph import Owners, OwnersNotFoundBehavior, OwnersRequest
+from pants.engine.rules import SubsystemRule, rule
 from pants.engine.selectors import Get, MultiGet
-from pants.engine.target import InferDependenciesRequest, InferredDependencies
+from pants.engine.target import (
+    HydratedSources,
+    HydrateSourcesRequest,
+    InferDependenciesRequest,
+    InferredDependencies,
+)
 from pants.engine.unions import UnionRule
+from pants.subsystem.subsystem import Subsystem
+
+
+class PythonInference(Subsystem):
+    """Options controlling which dependencies will be inferred for Python targets."""
+
+    options_scope = "python-infer"
```

Comment:
I think generally we have subsystems be nouns. I'd recommend `python-inference`.

Reply:
Good idea to have a dedicated subsystem, btw.

Reply:
I started with that, but the option names were so much nicer with the verb... `--python-infer-imports`, `--no-python-infer-imports`, etc. Should I change the option names as well?

Reply:
True, but this gets to Benjy's point that 90% of options are only ever set in `pants.toml`. I strongly suspect this will only be set in `pants.toml`, outside of a codebase admin experimenting with things.

So, it would read:

```toml
[python-inference]
imports = true
inits = true
conftests = true
```

Reply:
I kinda like `[python-infer]` even in config TBH

Reply:
Leaning toward leaving it then.